geoslurp_userplugins/sword.py

```diff
@@ -0,0 +1,84 @@
+# This file is part of geoslurp.
+# geoslurp is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 3 of the License, or (at your option) any later version.
+
+# geoslurp is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+
+# You should have received a copy of the GNU Lesser General Public
+# License along with geoslurp; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+
+# Author Roelof Rietbroek (r.rietbroek@utwente.nl), 2023
+# Author Amin Shakya (a.shakya@utwente.nl), 2023
+
+from geoslurp.dataset.OGRBase import OGRBase
+from geoslurp.datapull.http import Uri as http
+from geoslurp.datapull.github import Crawler as ghCrawler
+from geoslurp.datapull.github import GithubFilter as ghfilter
+from geoslurp.datapull.github import cachedGithubCatalogue
+from geoslurp.config.catalogue import geoslurpCatalogue
+from datetime import datetime
+import json
+from zipfile import ZipFile
+import os
+import tempfile
+import yaml
+
+class SwordBase(OGRBase):
+    """
+    Base class for SWORD: the SWOT River Database, Version 15. If you would like more information about SWORD, please see the paper published in Water Resources Research in 2021, led by Elizabeth Altenau. You can access it here: https://agupubs.onlinelibrary.wiley.com/doi/10.1029/2021WR030054
+    """
+    import os
```

Comment:
Hi Amin, the `import os` and `import zipfile` are not needed here, as they are already covered above

Reply:
@strawpants I have now deleted github related imports and the redundant `import os` and `import zipfile`. Request for a recheck.

Best,
AS